src/index.html

```diff
@@ -16,6 +16,17 @@
     <meta name="theme-color" content="white" />
     <link rel="manifest" href="/manifest.json" />
 
+    <!--
+      Create dataLayer for google analytics:
+      https://developers.google.com/tag-platform/tag-manager/web/datalayer
+    -->
+    <script>
+      window.dataLayer = window.dataLayer || [];
+      function gtag() {
+        dataLayer.push(arguments);
+      }
+    </script>
+
```

Comment:
I think this should moved back into your embed embedGoogleAnalytics function - there's really no reason to have it split from that file. Also when used with my suggestion about preembedding tags you should be able to avoid the retry loop altogether